File: kitware-prep/bank/bank_account.cpp
#include "bank_account.hpp"

// BankAccount::BankAccount(const std::string& pOwner, const double& pBalance)
// {
//     owner = pOwner;
//     banlance = pBalance;
// }
BankAccount::BankAccount(const std::string& pOwner, const double& pBalance) :
owner(pOwner), balance(pBalance)
{ 
}

std::string BankAccount::getOwner() const
{
    return owner;
}

double BankAccount::getBalance() const
{
    return balance;
}

void BankAccount::deposit(const double& pAmount)
{
    if(pAmount <= 0) return;

    balance += pAmount;
}

bool BankAccount::withdraw(const double& pAmount)
{
    if(balance < pAmount || pAmount <= 0) return false;

    balance -= pAmount;
    return true;
}

File: kitware-prep/bank-python/bank_account.py
class BankAccount:
    def __init__(self, owner, balance):
        self.owner, self.balance = owner, balance

    def display_balance(self):
        print(self.owner, "has", self.balance, "dollars")

    def deposit(self, amount):
        if amount < 0:
            return
        
        self.balance += amount

    def withdraw(self, amount):
        if amount > self.balance or amount < 0:
            return

        self.balance -= amount